File: src/main/java/alfheim/client/render/block/RenderBlockAnyavil.java
package alfheim.client.render.block;

import static org.lwjgl.opengl.GL11.*;

import alfheim.api.lib.LibRenderIDs;
import alfheim.common.block.tile.TileAnyavil;
import cpw.mods.fml.client.registry.ISimpleBlockRenderingHandler;
import net.minecraft.block.Block;
import net.minecraft.client.renderer.RenderBlocks;
import net.minecraft.client.renderer.tileentity.TileEntityRendererDispatcher;
import net.minecraft.world.IBlockAccess;

public class RenderBlockAnyavil implements ISimpleBlockRenderingHandler {

	@Override
	public void renderInventoryBlock(Block block, int metadata, int modelID, RenderBlocks renderer) {
		glPushMatrix();
		glRotated(-90, 0, 1, 0);
		glTranslated(-0.5, -0.6, -0.5);
		TileEntityRendererDispatcher.instance.renderTileEntityAt(new TileAnyavil(), 0, 0, 0, 0);
		glPopMatrix();
	}

	@Override
	public boolean renderWorldBlock(IBlockAccess world, int x, int y, int z, Block block, int modelId, RenderBlocks renderer) {
		return false;
	}

	@Override
	public int getRenderId() {
		return LibRenderIDs.idAnyavil;
	}

	@Override
	public boolean shouldRender3DInInventory(int modelId) {
		return true;
	}
}

File: src/main/java/alfheim/client/render/block/RenderBlockAnyavil.kt
package alfheim.client.render.block

import alfheim.api.lib.LibRenderIDs
import alfheim.common.block.tile.TileAnyavil
import cpw.mods.fml.client.registry.ISimpleBlockRenderingHandler
import net.minecraft.block.Block
import net.minecraft.client.renderer.RenderBlocks
import net.minecraft.client.renderer.tileentity.TileEntityRendererDispatcher
import net.minecraft.world.IBlockAccess
import org.lwjgl.opengl.GL11.*

object RenderBlockAnyavil: ISimpleBlockRenderingHandler {
	
	override fun renderInventoryBlock(block: Block, metadata: Int, modelID: Int, renderer: RenderBlocks) {
		glPushMatrix()
		glRotated(-90.0, 0.0, 1.0, 0.0)
		glTranslated(-0.5, -0.6, -0.5)
		TileEntityRendererDispatcher.instance.renderTileEntityAt(TileAnyavil(), 0.0, 0.0, 0.0, 0f)
		glPopMatrix()
	}
	
	override fun renderWorldBlock(world: IBlockAccess, x: Int, y: Int, z: Int, block: Block, modelId: Int, renderer: RenderBlocks) = false
	override fun shouldRender3DInInventory(modelId: Int) = true
	override fun getRenderId() = LibRenderIDs.idAnyavil
}